bluepill/src/BPPacker.m

```diff
@@ -16,11 +16,18 @@ @implementation BPPacker
 + (NSArray<BPXCTestFile *> *)packTests:(NSArray<BPXCTestFile *> *)xcTestFiles
                          configuration:(BPConfiguration *)config
                               andError:(NSError **)errPtr {
-    if (!config.testTimeEstimatesJsonFile) {
-        return [self packTestsByCount:xcTestFiles configuration:config andError:errPtr];
-    } else {
-        return [self packTestsByTime:xcTestFiles configuration:config andError:errPtr];
+    NSArray<BPXCTestFile *> *packedBundles = nil;
+    if (config.testTimeEstimatesJsonFile) {
+        packedBundles = [self packTestsByTime:xcTestFiles configuration:config andError:errPtr];
+        // Return if there is an error in packing by test time estimates
+        if (errPtr && *errPtr != nil)
+            return nil;
     }
+    // Split tests by count if time estimates json is mission or splitting by time estimates failed
+    if (!config.testTimeEstimatesJsonFile || !packedBundles) {
+        packedBundles = [self packTestsByCount:xcTestFiles configuration:config andError:errPtr];
```

Comment:
Shouldn't we also check `errPtr` here like above?

Reply:
The check would not be needed. If there is an error the bundles list would be nil and no further processing is required.